### DISCUSSION TITLE
PostGIS connection to QField

### DISCUSSION THREAD

**Discussion by careyshan** [opengisch/QField#2235] [Q&A]
Hi,
I'd like to do the direct connection to PostGIS. 
![image](https://user-images.githubusercontent.com/3117142/137932300-ba134cae-4d8f-4c14-901d-bf4c65205910.png)

What I don't understand is the Convert to configuration? And if I check the boxes, how are the credentials saved on the android device that is using QField? My worry is how secure is this? Is the username and password of PostGIS then exposed to anyone who has the device?

Thanks

**Comment by m-kuhn** [comment]
There is a difference if you are using plain QField or QField with QFieldCloud

But for both: You do not need to "convert to configuration", this is for QGIS only.

## QField

You have two possibilities:

1. You need to enter the credentials and "store" in the project file. They will be stored unencrypted in the file and anyone with access can access them.
2. You can also use a pg_service.conf file and place it on your device in a folder called QField. In this file you can define login credentials and this way not ship them with the project file.

It is not possible for a user to connect to the database without having login credentials somewhere.

## QFieldCloud

As QFieldCloud will make a copy of your data, these credentials will not leave the QFieldCloud server and only the data will be sent to the device. At the moment you need to go with possibility 1. (store credentials into to project file) but possibility 2. is likely to be implemented soon.

**Reply by careyshan** [reply]
Thanks @m-kuhn

**Reply by ldebomy** [reply]
Before 2.0.10, its works by putting pg_service.conf in /storage/emulated/0/QField. But since 2.0.10, I can't get it work.
I have tried :
 /storage/emulated/0/Android/data/ch.opengis.qfield/
 /storage/emulated/0/Android/data/ch.opengis.qfield/files/
 /storage/emulated/0/Android/data/ch.opengis.qfield/files/QField
without success.

**Reply by m-kuhn** [reply]
This had to be adjusted because of new Google Play policies https://www.opengis.ch/2022/03/05/qfield-users-sit-down-we-need-to-talk-about-storage-access-on-android%ef%bf%bc/
It needs to be adjusted analog to
https://docs.qfield.org/how-to/gnss/?h=#altitude-correction-vertical-grid-shift

Check your about screen for the appropriate location

![Screenshot_20220321-165255](https://user-images.githubusercontent.com/588407/159299810-38f0a6ba-ab4a-44a9-9308-bfc4f9a9298a.png)

**Reply by ldebomy** [reply]
Thank you for your answer. 
My Qfield directory is /storage/emulated/0/Android/data/ch.opengis.qfield/files/QField but putting pg_service.conf in it does not work. Log shows this message : Connection to database failed definition of service "mydb" not found.

**Reply by nirvn** [reply]
@Idebomy , do you have a QField directory in the root of your device's storage too by any chance?

**Reply by nirvn** [reply]
@ldebomy , you can try APKs here, hopefully it'll fix your issue: https://github.com/opengisch/QField/pull/2631#issuecomment-1074997353
